src/utils/framework.ts

```diff
@@ -67,6 +69,58 @@ const transformMatrix2dLevels = ({
     return dimensionFirstLevels;
 };
 
+const transformLevelsSectorFirst = ({
+    dimensions: widgetDim,
+    sectors: widgetSec,
+}: Matrix2dWidgetElement['properties']['data']) => {
+    const sectorFirstLevels = widgetSec.map((s) => {
+        const { subsectors } = s;
+        let sublevels: Level[] = [];
+        if (subsectors) {
+            sublevels = subsectors.map((ss) => {
+                const dimensions = widgetDim.map((d) => {
+                    const { subdimensions } = d;
+                    const subDimensionsLevel = subdimensions.map(sd => ({
+                        id: `${s.id}-${ss.id}-${d.id}-${sd.id}`,
+                        title: sd.title,
+                    }));
+                    return ({
+                        id: `${s.id}-${ss.id}-${d.id}`,
+                        title: d.title,
+                        sublevels: subDimensionsLevel,
+                    });
+                });
+                return ({
+                    id: `${s.id}-${ss.id}`,
+                    title: ss.title,
+                    sublevels: dimensions,
+                });
+            });
+        } else {
+            sublevels = widgetDim.map((d) => {
+                const { subdimensions } = d;
+                const subDimensionsLevel = subdimensions.map(sd => ({
+                    id: `${s.id}-${d.id}-${sd.id}`,
```

Comment:
Maybe we should create a simple function to do this repeated logic

```typescript
function createId<T extends string | number>(...args: [T, T, ...T[]]) {
    return args.join('-');
}
```

Reply:
This function will not accept null, undefined, or arguments with length less than 2.